### ISSUE TITLE
htest/test fails to build when confd is disabled

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#388]
Originally reported of Google Code with ID 332.

```
Building htest/test with “./configure --disable-confd” fails in a very non-obvious way. Either the target inMakefile should already fail with an error message, or at least the error generated by ghc should be more descriptive.

$ git describe
v2.6.1-899-g78fcfd4

$ ./autogen.sh && ./configure --prefix=/usr/local --sysconfdir=/etc --localstatedir=/var --enable-restricted-commands --disable-split-query --disable-confd && make -j2 htest/test
[…]
htools/Ganeti/Query/Filter.hs:128:9:
    No instance for (PCRE.RegexLike () [Char])
      arising from a use of `PCRE.match'
    Possible fix:
      add an instance declaration for (PCRE.RegexLike () [Char])
    In the second argument of `($!)', namely
      `PCRE.match (compiledRegex re) (fromJSString val)'
    In the expression:
      Ok $! PCRE.match (compiledRegex re) (fromJSString val)
    In an equation for `regexpFilter':
        regexpFilter re (JSString val)
          = Ok $! PCRE.match (compiledRegex re) (fromJSString val)

```
Originally added on 2012-12-10 15:04:37 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2013-03-11 13:19:07 +0000 UTC.

Added Labels: Priority-Medium

**Comment by Ganeti-Issues-Migrator** [comment]
```
(priority medium as it affects test code)
```
Originally added on 2013-03-11 13:19:27 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
We discussed this when adding confd/monitoring.

It is too complex to support all possible configure combinations in test files; it's hard enough to do it in production code. The compromise was that:

- we support building production code with various options enabled/disabled
- but test code only builds if all options are enabled, in effect requiring a 'full' Ganeti
```
Originally added on 2013-03-11 15:28:03 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
I agree with the solution, but the bug still exists: crashing during the build is not a good form, it should just refuse to build. :)

Thanks,

Guido
```
Originally added on 2013-03-11 15:51:49 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2013-04-10 14:56:41 +0000 UTC.

Added Labels: Priority-Low
Removed Labels: Priority-Medium

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2013-04-18 08:24:00 +0000 UTC.

Added to Milestone: Release2.8
Removed from Milestone: Release2.7

**Comment by Ganeti-Issues-Migrator** [comment]
```
Move non-critical bugs scheduled for 2.8 or 2.9 to 2.11, as in those versions only critical bug fixes will be integrated.
```
Originally added on 2013-10-30 09:48:16 +0000 UTC.

Added to Milestone: Release2.11
Removed from Milestone: Release2.8

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2013-10-30 10:19:53 +0000 UTC.

Added Labels: SmallTask

**Comment by Ganeti-Issues-Migrator** [comment]
```
Moving all issues that won't be addressed in 2.11 anymore to 2.12.
```
Originally added on 2014-05-14 11:28:36 +0000 UTC.

Added to Milestone: Release2.12
Removed from Milestone: Release2.11

**Comment by Ganeti-Issues-Migrator** [comment]
```
I am closing this issue because our test infrastructure has slightly changed since it was opened and I was not able to replicate the problem. This is according to our current team effort to migrate to Github issues. Please feel free to comment and re-open this if you think it's still relevant.

See https://groups.google.com/forum/#!topic/ganeti/b2-eukpsXpk for more details.
```
Originally added on 2016-12-15 11:01:55 +0000 UTC.

Changed State: Obsolete
Removed from Milestone: Release2.12
